packages/controller-utils/src/util.test.ts

```diff
@@ -436,57 +437,61 @@ describe('util', () => {
   describe('query', () => {
     describe('when the given method exists directly on the EthQuery', () => {
       it('should call the method on the EthQuery and, if it is successful, return a promise that resolves to the result', async () => {
-        class EthQuery {
+        class MockEthQuery extends EthQuery {
           getBlockByHash(blockId: any, cb: any) {
             cb(null, { id: blockId });
           }
         }
-        // @ts-expect-error Mock eth query does not fulfill type requirements
-        const result = await util.query(new EthQuery(), 'getBlockByHash', [
-          '0x1234',
-        ]);
+        const result = await util.query(
+          new MockEthQuery({} as Provider),
```

Comment:
Hmm. Now that we have a fake provider, I'm wondering if a better way to achieve this without the type assertion would be something like this:

``` suggestion
        const ethQuery = new EthQuery(new FakeProvider());
        jest.spyOn(ethQuery, 'getBlockByHash').mockResolvedValue({ id: '0x1234' });
        const result = await util.query(
          ethQuery,
```

(We'll have to import FakeProvider at the top of course)

Reply:
Is this the `FakeProvider` in `network-controller`? I'm getting this error:
```ts
error TS2345: Argument of type 'FakeProvider' is not assignable to parameter of type 'Provider'.
```
It's also currently not exported from `network-controller` so ideally we'd need to create a new release to add the export.

Reply:
Here's my attempt at applying `FakeProvider`: https://github.com/MetaMask/core/pull/2037/commits/0a036847c84399602c360943a2319d8f54ac5932

Reply:
I had thought that FakeProvider was defined at the top level like FakeBlockTracker. We'd have to pull it out to use like this, so never mind :(

Reply:
We could keep this PR open until that's done. `FakeProvider` seems like a definite improvement over `{} as Provider`.

Reply:
Okay, new PR created here: https://github.com/MetaMask/core/pull/2039

Reply:
Didn't realize it would be that simple.😅 Just added a review.

Reply:
I've merged #2039 so these changes should now work.

Reply:
Fixed all of the mock eth queries to use `FakeProvider`: https://github.com/MetaMask/core/pull/2037/commits/687d3eb0d633a825e9814ccf98d1b4aac1caf605